doc/Language/typesystem.pod6

```diff
@@ -926,17 +938,21 @@ are classes.
 
 For a full explanation see L<Grammars|/language/grammars>.
 
-=head3 Versioning and authorship
-
-Versioning and authorship can be applied via the adverbs
-X«C«:ver<>»|:ver<> (grammar)» and X«C«:auth<>»|:auth<> (grammar)».
-Both take a string as argument, for C<:ver> the string is converted to a
-L<Version|/type/Version> object. To query a grammars version and author use
-C<.^ver> and C<^.auth>.
-
-    grammar G:ver<4.2.3>:auth<me@here.local> {}
-    say [G.^ver, G.^auth];
-    # OUTPUT: «[v4.2.3 me@here.local]␤»
+=head3 Versioning, authorship, and API
+
+Versioning, authorship, and API can be applied via the adverbs
+X«C«:ver<>»|:ver<> (grammar)», X«C«:auth<>»|:auth<> (grammar)», and
```

Comment:
Don't really see a point in repeating things over for everything that can be versioned... I get this is the way it was before, but maybe as part of the overhaul we should centralize that information...

Reply:
A new section (right at the end of type system page) might the appropriate way of going about it. Like this:

```
=head1 Versioning, authorship, and API version.

When you declare a type, you can pass it a version, author, and/or API,
all of which you can subsequently introspect. The versioning, authorship, 
and/or API version of a type can be applied via the adverbs C«:ver<>»,
C«:auth<>», and C«:api<>» respectively.
All of them take a string as argument; for C<:ver> the string is converted to
a L«C«Version»|/type/Version» object, and for C<:api> the string is converted 
into an L<allomorph|/language/glossary#index-entry-Allomorph>
L«C«IntStr»|/type/IntStr» object. To query the version, author, and
API version of a type use C<.^ver>, C<.^auth>, and C<.^api> respectively,
as illustrated down below.

X«|:ver<> (class)»X«|:auth<> (class)»X«|:api<> (class)»
=head2 C<class>

    class R:ver<4.2.3>:auth<me@here.local>:api<1> {}
    say R.^ver;       # OUTPUT: «v4.2.3␤»
    say R.^ver.parts; # OUTPUT: «(4 2 3)␤»
    say R.^auth;      # OUTPUT: «me@here.local␤»
    say R.^api;       # OUTPUT: «1␤»

X«|:ver<> (role)»X«|:auth<> (role)»X«|:api<> (role)»
=head2 C<role>

    role R:ver<4.2.3>:auth<me@here.local>:api<1> {}
    say R.^ver;       # OUTPUT: «v4.2.3␤»
    say R.^ver.parts; # OUTPUT: «(4 2 3)␤»
    say R.^auth;      # OUTPUT: «me@here.local␤»
    say R.^api;       # OUTPUT: «1␤»

X«|:ver<> (module)»X«|:auth<> (module)»X«|:api<> (module)»
=head2 C<module>

    module M:ver<4.2.3>:auth<me@here.local>:api<1> {}
    say M.^ver;       # OUTPUT: «v4.2.3␤»
    say M.^ver.parts; # OUTPUT: «(4 2 3)␤»
    say M.^auth;      # OUTPUT: «me@here.local␤»
    say M.^api;       # OUTPUT: «1␤»

X«|:ver<> (grammar)»X«|:auth<> (grammar)»X«|:api<> (grammar)»
=head2 C<grammar>

    grammar G:ver<4.2.3>:auth<me@here.local>:api<1> {}
    say G.^ver;       # OUTPUT: «v4.2.3␤»
    say G.^ver.parts; # OUTPUT: «(4 2 3)␤»
    say G.^auth;      # OUTPUT: «me@here.local␤»
    say G.^api;       # OUTPUT: «1␤»

```

Reply:
This is probably much better... You could use different calls, also, not clear the value in repeating the same calls over and over...

Reply:
Also, about indexing... Not a lot of value putting so many indices close together, and pointing to examples. A single index without categories would probably do.